OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -348,12 +401,31 @@ bool LoadAsset(IReadOnlyPackage package, string filename)
 					return true;
 				}
 
-				currentSprites = world.Map.Rules.Sequences.SpriteCache[prefix + filename];
-				currentFrame = 0;
-				if (frameSlider != null)
+				if (Path.GetExtension(filename.ToLowerInvariant()) == ".vxl")
+				{
+					var voxelName = Path.GetFileNameWithoutExtension(filename);
+					var modelSequences = world.Map.Rules.ModelSequences;
+
+					currentVoxel = null;
+					if (modelSequences.ContainsKey(voxelName))
+						currentVoxel = world.ModelCache.GetModelSequence(voxelName, "idle");
```

Comment:
This is really fragile, and generally a bad idea. There is no reason to expect that mods will always name their model images based on the filename of the voxel, or that the sequence will have an "idle" sequence. Even TS's small number of definitions has several examples that break these assumptions.

The asset browser does not make these assumptions for sprite sequences or vqas, so it shouldn't be making them for models either - please just use `GetModel` for all vxl files.

It also seems a bit odd that this doesn't use `prefix + filename` like the other cases.

Reply:
I agree that this should magic bytes if they exist, but you did the same assumptions in the engine

https://github.com/OpenRA/OpenRA/blob/9c4fd0e3d37f64b0c426b152ab364a6baa0ca51e/OpenRA.Mods.Cnc/Graphics/VoxelLoader.cs#L217-L220

Reply:
Uh, that is completely different to what I am saying here?